plugins/Kaleidoscope-DynamicMacros/src/kaleidoscope/plugin/DynamicMacros.cpp

```diff
@@ -272,14 +275,14 @@ EventHandlerResult DynamicMacros::onFocusEvent(const char *command) {
     } else {
       uint16_t pos = 0;
 
-      while (!::Focus.isEOL()) {
+      while (!::Focus.isEOL() && pos < storage_size_) {
         uint8_t b;
         ::Focus.read(b);
 
         Runtime.storage().update(storage_base_ + pos++, b);
       }
       Runtime.storage().commit();
```

Comment:
I don't know what the consequences are of not calling `commit()` after `update()`, but if it works the way I guess it does, this might be a good thing to add here:
```c++
if (pos > storage_size_) {
  return EventHandlerResult::EVENT_CONSUMED;
}
```

Reply:
(I meant _above_ line 284, not below it)